fuzztests/fuzzer.fuzz.js

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-undef, @typescript-eslint/no-var-requires */
+
+const { fuzzer } = require("@jazzer.js/fuzzer");
+const { FuzzedDataProvider } = require("@jazzer.js/core");
+
+describe("fuzzer", () => {
+	it.fuzz("traceStrCmp", (data) => {
+		const provider = new FuzzedDataProvider(data);
+		let a = provider.consumeString(10);
+		let b = provider.consumeString(10);
+		let op = provider.consumeString(5);
+		expect(fuzzer.traceStrCmp(a, b, op, 0));
+	});
+
+	it.fuzz("use never zero policy", (data) => {
+		const provider = new FuzzedDataProvider(data);
+		const iterations = provider.consumeIntegralInRange(1, 1 << 16);
+		for (let i = 0; i < iterations; i++) {
+			fuzzer.incrementCounter(0);
```

Comment:
The fuzzing of internal modules looks suspicious (see the screenshot below).
While coverage stays the same, the features are continuously increasing (always +1).
I think, `incrementCounter` should not be manipulating the counters of the fuzzing fuzzer---by manipulating the coverage  it converts our fuzzer into a non-coverage-guided random fuzzer.

![image](https://user-images.githubusercontent.com/240414/211059595-eaa0a6d0-daaf-4ecd-b708-f6436f707f30.png)

Reply:
Do you think instrumenting the fuzzer code in general or only the counter handling should be avoided?
Furthermore, executing the loop "more times" should already lead to an increase of features, shouldn't it?

Reply:
It shouldn't and it doesn't :) For some reason I was looking at the code in fuzzer.fuzz.js, however the screenshot is from the FuzzedDataProvider.fuzz.js. The fuzzer fuzz tests reach certain coverage rather quickly and are stuck there for the rest of the test. So, forget everything I wrote!